packages/web-components/src/components/ic-dialog/ic-dialog.tsx

```diff
@@ -0,0 +1,385 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Host,
+  Prop,
+  State,
+  Method,
+  Listen,
+  h,
+} from "@stencil/core";
+import closeIcon from "../../assets/close-icon.svg";
+import { isSlotUsed } from "../../utils/helpers";
+
+@Component({
+  tag: "ic-dialog",
+  styleUrl: "ic-dialog.css",
+  shadow: true,
+})
+export class Dialog {
+  /**
+   * @slot - Content will be placed in the middle of the dialog, between the heading and the controls.
+   * @slot dialog-controls - Content will be place at the bottom of the dialog.
+   */
+  @Element() el: HTMLIcDialogElement;
+
+  /**
+   * Sets the maximum and minimum height and width for the dialog.
+   */
+  @Prop() size?: "small" | "medium" | "large" = "small";
+  /**
+   * Sets the heading for the dialog.
+   */
+  @Prop() heading!: string;
+  /**
+   * Sets the optional label for the dialog which appears above the heading.
+   */
+  @Prop() label?: string;
+  /**
+   * Sets the dismiss label tooltip and aria label, default is 'dismiss'
+   */
+  @Prop() dismissLabel?: string = "Dismiss";
+  /**
+   * If set displays an alert of the corresponding variant below the heading.
+   */
+  @Prop() status?: "neutral" | "info" | "warning" | "error" | "success";
+  /**
+   * If a status is set, sets the heading for the displayed alert.
+   */
+  @Prop() alertHeading?: string;
+  /**
+   * If a status is set, sets the message for the displayed alert.
+   */
+  @Prop() alertMessage?: string;
+  /**
+   * Sets the number of default buttons to be displayed at the bottom of the dialog.
+   */
+  @Prop() buttons?: "none" | "one" | "two" | "three" = "two";
```

Comment:
Should the number of buttons be based on the number of buttonProps (with a max of 3)? That way, every button prop is account for. There will never be an instance where the buttons prop above is 1 but there are 3 buttonProps.

A .`map` on line https://github.com/mi6/ic-ui-kit/pull/438/files#diff-262ef4802b44b3bf5d24bf0350b75541588fc233edfa4ff24bb846bc3330670fR328 would do the work for you..